src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -284,7 +284,9 @@ private bool HasJobAttribute(CustomAttributeData attributeData)
 
             if (triggerBinding != null)
             {
-                functionDefinition = CreateTriggeredFunctionDefinition(triggerBinding, triggerParameterName, _executor, functionDescriptor, nonTriggerBindings, invoker, _singletonManager);
+                Type triggerValueType = triggerBinding.TriggerValueType;
+                MethodInfo createTriggeredFunctionDefinitionMethodInfo = typeof(FunctionIndexer).GetMethod("CreateTriggeredFunctionDefinitionImpl", BindingFlags.Instance | BindingFlags.NonPublic).MakeGenericMethod(triggerValueType);
```

Comment:
Might as well rename CreateTriggeredFunctionDefinitionImpl by removing the "Impl" suffix?